linkerd/meshtls/build.rs

```diff
@@ -0,0 +1,16 @@
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Ensure that at least one TLS implementation feature is enabled.
+    static TLS_FEATURES: &[&str] = &["rustls"];
+    if !TLS_FEATURES
+        .iter()
+        .any(|f| std::env::var_os(&*format!("CARGO_FEATURE_{}", f.to_ascii_uppercase())).is_some())
+    {
+        return Err(format!(
+            "at least one of the following TLS implementations must be enabled: '{}'",
+            TLS_FEATURES.join("', '"),
+        )
+        .into());
+    }
+
+    Ok(())
```

Comment:
FWIW, I think this could be achieved without a build script by just sticking
```rust
#![cfg(not(any(feature = "rustls"))]
compile_error!("at least one of the following TLS implementations must be enabled: 'rustls}''")
```
in `lib.rs` (and adding the other implementation feature flags as needed)